tests/utils/test_downstream.py

```diff
@@ -0,0 +1,134 @@
+
+
+@mock_api(enforce_online_test=ONLINE_TEST)
+def test_get_dataset_downstream_dependencies_suggest_name():
+    client = tamr_toolbox.utils.client.create(**CONFIG["toolbox_test_instance"])
+    dataset = client.datasets.by_resource_id(MASTERING_UNIFIED_DATASET_ID)
+    downstream_datasets = tamr_toolbox.utils.downstream.datasets(
+        dataset, include_dependencies_by_name=True
+    )
+    # check returned results
+    # in newer version of Tamr, this number might increase
+    # because more persistence data is moved to HBase
+    assert len(downstream_datasets) >= 49
+    DEPENDENT_DATASETS_BY_NAME = {
+        "minimal_mastering_unified_dataset_dedup_verified_cluster_members",
+        "minimal_mastering_unified_dataset.recordPairLabel",
+        "minimal_mastering_unified_dataset.record_pair_feedback_uid",
+        "minimal_mastering_unified_dataset.record_pair_id_string",
+        "minimal_mastering_unified_dataset.internal_links",
+        "minimal_mastering_unified_dataset_dedup_suggested_clusters_log",
+        "minimal_mastering_unified_dataset.cluster_feedback",
+        "minimal_mastering_unified_dataset.internal_comments",
+        "minimal_mastering_unified_dataset.global_record_id",
+    }
+    assert DEPENDENT_DATASETS_BY_NAME.issubset(set([d.name for d in downstream_datasets]))
+
+
+@mock_api(enforce_online_test=ONLINE_TEST)
+def test_get_downstream_projects():
+    client = tamr_toolbox.utils.client.create(**CONFIG["toolbox_test_instance"])
+    dataset = client.datasets.by_resource_id(MASTERING_UNIFIED_DATASET_ID)
+    downstream_projects = tamr_toolbox.utils.downstream.projects(
+        dataset, include_dependencies_by_name=True
+    )
+    assert len(downstream_projects) == 2
+    assert (
+        str(downstream_projects) == "[tamr_unify_client.project.resource.Project(relative_id="
+        "'projects/1', name='minimal_mastering', "
+        "type='DEDUP'), tamr_unify_client.project.resource.Project"
+        "(relative_id='projects/2', name='minimal_golden_records', "
+        "type='GOLDEN_RECORDS')]"
+    )
+
+
+@mock_api(enforce_online_test=ONLINE_TEST)
+def test_get_downstream_datasets_for_source_dataset():
+    client = tamr_toolbox.utils.client.create(**CONFIG["toolbox_test_instance"])
+    source_dataset = client.datasets.by_resource_id(SOURCE_DATASET_ID)
+    unified_dataset = client.datasets.by_resource_id(MASTERING_UNIFIED_DATASET_ID)
+    source_downstream_datasets = tamr_toolbox.utils.downstream.datasets(source_dataset)
+    unified_downstream_datasets = tamr_toolbox.utils.downstream.datasets(unified_dataset)
+    source_downstream_datasets_names = set([d.name for d in source_downstream_datasets])
+    unified_downstream_datasets_names = set([d.name for d in unified_downstream_datasets])
+    # All downstream datasets of unified dataset must be downstream datasets of the source dataset
+    assert unified_downstream_datasets_names.issubset(source_downstream_datasets_names)
```

Comment:
Same question as before, why are we only checking for a subset?

Reply:
list of downstream datasets of unified dataset should be a subset of downstream datasets of a source dataset, not equal